src/watts/plugin_pyarc.py

```diff
@@ -0,0 +1,186 @@
+    inputs
+        List of input files
+    outputs
+        List of output files
+
+    Attributes
+    ----------
+
+    """
+
+    def __init__(self, params: Parameters, time: datetime,
+                 inputs: List[Path], outputs: List[Path], user_object):
+        super().__init__('PyARC', params, time, inputs, outputs, user_object)
+        self.results_data = self._save_PyARC(user_object)
+
+    @property
+    def stdout(self) -> str:
+        return (self.base_path / "PyARC_log.txt").read_text()
+
+    def _save_PyARC(self, user_object) -> dict:
+        """Return PyARC results in a dictionary
+
+        Returns
+        -------
+        Results from PyARC results
+
+        """
+        results_data = {}
+        results_data["mcc3"] = user_object.results_keff_mcc3
+        results_data["dif3d"] = user_object.results_keff_dif3d
+        results_data["proteus_nodal"] = user_object.results_keff_nodal
+        results_data["rzmflx"] = user_object.results_keff_rzmflx
+        results_data["persent_pert"] = user_object.results_kpert_persent
+        results_data["persent_sens"] = user_object.results_kuq_persent
+        results_data["gamsor"] = user_object.results_power_gamsor
+        results_data["dassh"] = user_object.results_dassh
+        # TODO - this is minimum information that can be brought back easily. 
+        # More should be returned but will require work on PyARC.
+        return results_data
+
+    def save(self, filename: PathLike):
+        """Save results to an HDF5 file
+
+        Parameters
+        ----------
+        filename
+            File to save results to
+        """
+        with h5py.File(filename, 'w') as h5file:
+            super()._save(h5file)
+
+    @classmethod
+    def _from_hdf5(cls, obj: h5py.Group):
+        """Load results from an HDF5 file
+
+        Parameters
+        ----------
+        obj
+            HDF5 group to load results from
+        """
+        time, parameters, inputs, outputs = Results._load(obj)
+        return cls(parameters, time, inputs, outputs)
+
+class PluginPyARC(TemplatePlugin):
+    """Plugin for running PyARC
+
+    Parameters
+    ----------
+    template_file
+        Templated PyARC input
+    show_stdout
+        Whether to display output from stdout when SAM is run
+    show_stderr
+        Whether to display output from stderr when SAM is run
+    supp_inputs
+        List of supplementary input files that are needed for running PyARC
+    
+    Attributes
+    ----------
+    pyarc_exec
+        Path to PyARC executable
+
+    """
+
+    def  __init__(self, template_file: str, show_stdout: bool = False,
+                  show_stderr: bool = False, supp_inputs: List[str] = []):
+        super().__init__(template_file)
+        self._pyarc_exec = Path('PyARC.py')
+#        if os.environ["PyARC_DIR"] is not None: # how would I setup this as default exec if not specified?
+#            self._pyarc_exec = os.environ["PyARC_DIR"]
```

Comment:
This should do what you want:
```suggestion
        self._pyarc_exec = Path(os.environ.get('PyARC_DIR', 'PyARC.py'))
```